backend/middlewares/authentication.js

```diff
@@ -0,0 +1,33 @@
+import models from '../database/models'
+import * as crypto from 'crypto'
+
+export class Authentication {
+
+    static generateAuthToken(username, password) {
+        try {
+            const hashInput = username + password;
+            const hash = crypto.createHash('sha256');
+            hash.update(hashInput);
+            const authToken = hash.digest('hex');
+            return authToken;
+        } catch (error) {
+            throw new Error("Unexpected error generating auth token\n" + error)
+        }
+    }
```

Comment:
`generateAuthToken` should take the whole user object as a parameter. So in ` /users/authenticate` you would first make sure the user exists in the db, and then if they exist you can generate an authToken from the user object that you retrieve from the db